barretenberg/cpp/src/barretenberg/stdlib_circuit_builders/goblin_ultra_circuit_builder.cpp

```diff
@@ -12,7 +12,10 @@ namespace bb {
 
 template <typename FF> void GoblinUltraCircuitBuilder_<FF>::finalize_circuit()
 {
-    UltraCircuitBuilder_<UltraHonkArith<FF>>::finalize_circuit();
+    if (!this->circuit_finalized) {
+        add_gates_to_ensure_all_polys_are_non_zero();
```

Comment:
maybe should change this to add_goblin_ultra_gates_.... to be clear that this only adds the goblin_ultra gates and not the ultra ones anymore

Reply:
done